app/controllers/resources_controller.rb

```diff
@@ -2,35 +2,16 @@ class ResourcesController < ApplicationController
   def index
     category_id = params[:category_id]
     result = if category_id
-               resources.joins(:categories).where('categories.id' => category_id)
```

Comment:
The `resources` method is doing the prefetching to avoid O(n) queries (but as I mentioned tonight, it's not prefetching enough). We should probably add it back, and I'll what's missing in my next PR.